cmake/generic.cmake

```diff
@@ -78,6 +78,10 @@
 #
 #   cc_test(example_test SRCS example_test.cc DEPS example glog gflags)
 
+if(WITH_GPU)
+  add_definitions(-DPADDLE_WITH_GPU)
```

Comment:
Do you mean [`PADDLE_ONLY_CPU`](https://github.com/PaddlePaddle/Paddle/blob/v0.10.0/cmake/configure.cmake#L44)? OK. switch to use it.